src/Mollie/WC/Helper/ApplePayDataObjectHttp.php

```diff
@@ -0,0 +1,470 @@
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay on update shipping contact
+     * Required data depends on callerPage
+     * @param array $post
+     */
+    public function updateContactData(array $post)
+    {
+        $this->resetErrors();
+        $requiredFields = self::UPDATE_CONTACT_SINGLE_PROD_REQUIRED_FIELDS;
+        if (array_key_exists('callerPage', $post)
+            && $post['callerPage'] == 'cart'
+        ) {
+            $requiredFields = self::UPDATE_CONTACT_CART_REQUIRED_FIELDS;
+        }
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            $requiredFields
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+        $this->simplifiedContact = $this->simplifiedAddress(
+            filter_var_array(
+                $post['simplifiedContact'],
+                FILTER_SANITIZE_STRING
+            )
+        );
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay on update shipping method
+     * Required data depends on callerPage
+     *
+     * @param array $post
+     */
+    public function updateMethodData(array $post)
+    {
+        $this->resetErrors();
+        $requiredFields = self::UPDATE_METHOD_SINGLE_PROD_REQUIRED_FIELDS;
+
+        if (array_key_exists('callerPage', $post)
+            && $post['callerPage'] == 'cart'
+        ) {
+            $requiredFields = self::UPDATE_METHOD_CART_REQUIRED_FIELDS;
+        }
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            $requiredFields
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+
+        $this->simplifiedContact = $this->simplifiedAddress(
+            filter_var_array(
+                $post['simplifiedContact'],
+                FILTER_SANITIZE_STRING
+            )
+        );
+        $this->shippingMethod = filter_var_array(
+            $post['shippingMethod'],
+            FILTER_SANITIZE_STRING
+        );
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay on authorized order
+     * Required data depends on callerPage
+     *
+     * @param array $post
+     * @param       $callerPage
+     */
+    public function orderData(array $post, $callerPage)
+    {
+        $this->resetErrors();
+        $requiredFields = self::CREATE_ORDER_SINGLE_PROD_REQUIRED_FIELDS;
+        if ($callerPage == 'cart') {
+            $requiredFields = self::CREATE_ORDER_CART_REQUIRED_FIELDS;
+        }
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            $requiredFields
+        )
+        ) {
+            return;
+        }
+
+        $this->assignDataObjectValues($post);
+
+
+        if (!array_key_exists('emailAddress', $post['shippingContact'])
+            || !$post['shippingContact']['emailAddress']
+        ) {
```

Comment:
Ok but where do you ensure `$post['shippingContact']` ?

Reply:
shippingContact is one of the required fields for this method, so in the case it doesn't exist updateRequiredData would return false and would add an error to the list of errors (that later are sent to Apple for the user to correct). On updateRequiredData false we return, so if we arrive to that point the shippingContact is there